code/src/Models/User.ts

```diff
@@ -2,58 +2,90 @@ import app from "firebase"
 
 
 export default class User{
+    /**
+     * User class contains uid, firstName, lastName
+     * @author Mohamad Abdel Rida
+     * 
+     * 
+     */
+
+
     firstName?:string
     lastName?:string
+    email:string
     uid:string
-    // TODO add other attributes
-    constructor(uid:string,firstName?:string, lastName?:string){
+
+    /**
+     * 
+     * @param uid user id used in firebase
+     * @param firstName user's first name | from firestore or user input
+     * @param lastName user's last name | from firestore user input
+     * @param email user's email address
+     */
+
+    constructor(uid:string,email:string,firstName?:string, lastName?:string){
         this.firstName = firstName
         this.lastName =lastName
         this.uid = uid
+        this.email = email
     }
 
     private async addToDb(){
```

Comment:
Should we return a Promise here? Otherwise the invoker doesn't know the result of what happened

Reply:
Promise on the constructor? I was thinking we could throw an error if fname and lname were null on fetchUserDetails()

Reply:
My bad haha, I thought I was reading further down the file lol. Ignore the promise bit here. I meant to write this on fetchUserDetails()